File: app/src/main/java/com/easydictionary/app/core/translate/yandex/YandexDictionaryResult.java
package com.easydictionary.app.core.translate.yandex;

import com.easydictionary.app.core.translate.DictionaryResult;
import com.google.gson.annotations.SerializedName;

public class YandexDictionaryResult implements DictionaryResult {

    @SerializedName("def")
    private Definition[] definitions;

    @Override
    public String toString() {
        return takeResult();
    }

    @Override
    public String takeResult() {
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < definitions.length; i++) {
            sb.append(i + 1).append(") ").append(definitions[i].toString());
            if (i != definitions.length - 1) {
                sb.append("\n");
            }
        }
        return sb.toString();
    }
}

File: app/src/main/kotlin/org/book2words/translate/yandex/YandexDictionaryResult.kt
package org.book2words.translate.yandex

import com.google.gson.annotations.SerializedName
import org.book2words.translate.core.DictionaryResult

class YandexDictionaryResult : DictionaryResult {
    SerializedName("def")
    private val definitions: Array<Definition>? = null

    override fun results(): Array<out org.book2words.translate.core.Definition>? {
        return definitions
    }

}
